src/makeInfinityAdder.js

```diff
@@ -4,7 +4,21 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let total = 0;
+
+  return function adder(value) {
+    if (value === undefined) {
+      const result = total;
+
+      total = 0;
+
+      return result;
+    } else {
```

Comment:
```suggestion
    }
```
else is redundant here